assets/tests/hooks/useRouteFilter.test.tsx

```diff
@@ -0,0 +1,80 @@
+import { act } from "react-dom/test-utils"
+import useRouteFilter, {
+  filterRoutes,
+  RouteFilter,
+} from "../../src/hooks/useRouteFilter"
+import { Route } from "../../src/skate"
+import { testHook } from "../testHelpers/testHook"
+
+let routeFilter: RouteFilter
+
+describe("useRouteFilter", () => {
+  beforeEach(() => {
+    testHook(() => {
+      // tslint:disable-next-line: react-hooks-nesting
+      routeFilter = useRouteFilter()
+    })
+  })
+
+  test("returns a RouteFilter", () => {
+    expect(typeof routeFilter.filterType).toBe("string")
+    expect(typeof routeFilter.filterText).toBe("string")
+    expect(routeFilter.handleTypeChange).toBeInstanceOf(Function)
+    expect(routeFilter.handleTextInput).toBeInstanceOf(Function)
+    expect(routeFilter.clearTextInput).toBeInstanceOf(Function)
+  })
```

Comment:
Is this necessary? The typescript compiler should take care of this.

Reply:
No, not necessary, I can remove it.